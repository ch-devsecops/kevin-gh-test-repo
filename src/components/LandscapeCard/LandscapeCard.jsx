import React from 'react';
import PropTypes from 'prop-types';
import {
  Wrapper,
  Row,
  Box,
  MediaItem,
  Optional,
  Markdown,
  MarkdownHeading,
  Media,
  useOptionalVideo,
} from '@honda-canada/design-system-react';
import themeStyles from './LandscapeCard.styles';

const Container = themeStyles.apply(Row, 'Container');
const Content = themeStyles.apply(Box, 'Content');
const BodyTextContainer = themeStyles.apply(Box, 'BodyTextContainer');
const CTAGroup = themeStyles.apply(Box, 'CTAGroup');
const StyledMedia = themeStyles.apply(Media, 'Media');
const MediaContainer = themeStyles.apply(Box, 'MediaContainer');

// Renders a Media Component that either renders an image or video
const MediaContent = ({ image, video, inSideNav, ...rest }) => (
  <StyledMedia {...rest}>
    <MediaContainer inSideNav={inSideNav}>
      <MediaItem image={image} video={video} />
    </MediaContainer>
  </StyledMedia>
);

MediaContent.propTypes = {
  image: PropTypes.element,
  video: PropTypes.shape({
    onPlay: PropTypes.func,
    ariaLabel: PropTypes.string,
    closeAriaLabel: PropTypes.string,
  }),
  inSideNav: PropTypes.bool,
};

const LandscapeCard = ({
  title,
  bodyText,
  ctas,
  image,
  video,
  contentAlignment,
  backgroundColor,
  isDark,
  gtmTags = {},
  layoutContainer,
  anchorId,
  ...otherProps
}) => {
  const { optionalVideo } = useOptionalVideo(video);

  const inSideNav = layoutContainer === 'SideNavLayout';

  return (
    <Box
      id={anchorId}
      bg={isDark ? 'black' : backgroundColor}
      data-gtm-category={gtmTags.category}
      data-gtm-component-type={gtmTags.type}
      pb={['xl', 'xxl']}
      pt={['xl', 'xxl']}
      {...otherProps}
    >
      <Wrapper>
        <Container contentAlignment={contentAlignment}>
          <Content contentAlignment={contentAlignment}>
            <MediaContent at="mobile" image={image} video={video} />

            <Optional when={title}>
              <MarkdownHeading
                style={{ textAlign: ['center', 'left'] }}
                headingOverride="h3"
                color={isDark ? 'white' : ''}
              >
                {title}
              </MarkdownHeading>
            </Optional>

            <Optional when={bodyText}>
              <BodyTextContainer>
                <Markdown color={isDark ? 'white' : ''}>{bodyText}</Markdown>
              </BodyTextContainer>
            </Optional>

            <Optional when={ctas?.length}>
              <CTAGroup>{ctas}</CTAGroup>
            </Optional>
          </Content>

          <MediaContent greaterThan="mobile" image={image} video={optionalVideo} inSideNav={inSideNav} />
        </Container>
      </Wrapper>
    </Box>
  );
};

LandscapeCard.defaultProps = {
  contentAlignment: 'left',
  backgroundColor: 'white',
};

LandscapeCard.propTypes = {
  /**
   * Array of React Components or HTML elements
   */
  ctas: PropTypes.arrayOf(PropTypes.element),
  title: PropTypes.string,
  image: PropTypes.element,
  video: PropTypes.shape({
    src: PropTypes.string,
    ariaLabel: PropTypes.string,
    closeAriaLabel: PropTypes.string,
    onPlay: PropTypes.func,
  }),
  /**
   * Markdown driven element
   */
  bodyText: PropTypes.string,
  /**
   * Orientation of Text within Card
   */
  contentAlignment: PropTypes.oneOf(['left', 'right']),
  /**
   * Sets background color of Container
   */
  backgroundColor: PropTypes.oneOf(['white', 'grey.5']),
  gtmTags: PropTypes.shape({
    category: PropTypes.string,
    type: PropTypes.string,
  }),
  layoutContainer: PropTypes.string,
  isDark: PropTypes.bool,
  anchorId: PropTypes.string,
};

export default LandscapeCard;
