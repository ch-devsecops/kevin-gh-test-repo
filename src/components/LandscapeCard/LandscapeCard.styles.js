import css from '@styled-system/css';
import variant from '@styled-system/variant';
import { ThemeStyles } from '@honda-canada/design-system-react';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('Container')(({ contentAlignment }) => ({
  height: '100%',
  flexDirection: contentAlignment === 'right' && 'row-reverse',
  alignItems: ['top'],
}));

themeStyles.addBaseStyles('Media')(
  css({
    width: ['100%', '50%'],
  }),
);

themeStyles.addBaseStyles('MediaContainer')(({ inSideNav }) =>
  css({
    px: ['0', inSideNav ? '0' : 'columnGap.1'],
    height: ['150px', inSideNav ? 'auto' : '350px'],
    maxHeight: inSideNav && '255px',
    maxWidth: inSideNav && '453px',
    mt: ['default', 0],
    mb: ['xs', 0],
  }),
);

themeStyles.addBaseStyles('Content')(
  {
    display: 'flex',
    flexDirection: 'column',
  },

  css({
    justifyContent: ['initial', 'center'],
    width: ['100%', '50%'],

    '& >:last-child': {
      mb: 0,
    },
  }),

  variant({
    prop: 'contentAlignment',
    variants: {
      left: {
        pr: ['columnGap.0', '52px'],
        pl: ['columnGap.0', 'columnGap.1'],
      },
      right: {
        pr: ['columnGap.0', 'columnGap.1'],
        pl: ['columnGap.0', '52px'],
      },
    },
  }),
);

themeStyles.addBaseStyles('BodyTextContainer')(
  css({
    my: ['xs', 'm'],
    p: {
      mb: ['default'],
    },
    h6: {
      mb: ['4px'],
      color: ['blue'],
    },
  }),
);

themeStyles.addBaseStyles('CTAGroup')(
  css({
    display: 'flex',
    flexWrap: 'wrap',
    flexDirection: ['column', 'row'],
    alignItems: 'center',
    alignContent: ['center', 'flex-start'],

    '& >:first-child': {
      mr: [0, 'm'],
    },

    '& > *': {
      mt: 'm',
    },
  }),
);

export default themeStyles;
