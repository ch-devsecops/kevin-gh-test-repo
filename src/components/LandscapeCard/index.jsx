import React, { useContext } from 'react';
import { Image, H1 } from '@honda-canada/design-system-react';
import LandscapeCardReact from './LandscapeCard';
import { LayoutContext } from '../LayoutContext';
import { ModelExplorationContext } from '../ModelExplorationContext';
import { getVideoProps, mapGTMCategory, mapJssFieldsToCtaComponents } from '../../utils/sitecoreFields';
import { getGtmTagValue } from '../../utils/gtmEvents';
import { stripMarkdownHeading, styledCompiler } from '../../utils/markdown';

const LandscapeCard = ({ fields, params, rendering }) => {
  const modelExplorationContext = useContext(ModelExplorationContext) || {};
  const { isDark } = modelExplorationContext;
  const { layoutName } = useContext(LayoutContext);
  if (!fields) return null;
  const { anchorId, title, bodyText, mediaImage, gtmTitle, gtmCategory, videoUrl, seoH1 } = fields;
  const inSideNav = layoutName === 'SideNavLayout';
  const imageStyle = inSideNav ? { objectFit: 'contain' } : {};
  const ctas = mapJssFieldsToCtaComponents(fields, gtmTitle?.value, rendering?.componentName);
  const gtmTags = {
    category: mapGTMCategory(gtmCategory),
    title: getGtmTagValue(gtmTitle?.value),
    type: getGtmTagValue(rendering?.componentName),
  };

  const generateH1Component = (fieldValue, key) =>
    key === seoH1?.value ? <H1 as="h1">{styledCompiler(stripMarkdownHeading(fieldValue))}</H1> : fieldValue;

  return (
    <LandscapeCardReact
      layoutContainer={layoutName}
      title={generateH1Component(title?.value, 'Title')}
      bodyText={generateH1Component(bodyText?.value, 'Description')}
      ctas={ctas}
      anchorId={anchorId?.value}
      gtmTags={gtmTags}
      image={<Image {...mediaImage?.value} style={imageStyle} />}
      video={getVideoProps(videoUrl?.value)}
      backgroundColor={params?.contentBgColour === 'Grey' ? 'grey.5' : 'white'}
      contentAlignment={params?.contentAlignment?.toLowerCase()}
      pt={params?.topMargin ? `${params.topMargin}px` : ['xl', 'xxl']}
      pb={params?.bottomMargin ? `${params.bottomMargin}px` : ['xl', 'xxl']}
      isDark={isDark}
    />
  );
};

export default LandscapeCard;
