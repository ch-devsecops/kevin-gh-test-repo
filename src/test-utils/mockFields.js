/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
