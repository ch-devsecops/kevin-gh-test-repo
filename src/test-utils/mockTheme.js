/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
