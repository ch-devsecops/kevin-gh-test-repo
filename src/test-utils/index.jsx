import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
