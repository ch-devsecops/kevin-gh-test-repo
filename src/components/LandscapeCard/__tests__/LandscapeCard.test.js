import React from 'react';
import { axe, cleanup, render, MockRendering, MockFields } from '../../../test-utils';
import LandscapeCard from '../LandscapeCard';

afterEach(cleanup);

const mockFields = new MockFields();
const mockRendering = new MockRendering('LandscapeCard');

describe('<LandscapeCard>', () => {
  it('should render without accessibility violations', async () => {
    const { container } = render(
      <LandscapeCard fields={mockFields} rendering={mockRendering} title="My title" bodyText="My body text" />,
    );
    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
