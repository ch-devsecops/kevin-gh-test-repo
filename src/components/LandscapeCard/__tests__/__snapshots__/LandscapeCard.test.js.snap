// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandscapeCard> should render without accessibility violations 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 hgeMfw"
  >
    <div
      class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 gXilRl"
    >
      <div
        class="Box-sc-1bsf784-0 jGVkdm sc-1k44dfv-0 bTFGsD"
        display="flex"
        width="100%"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 fZdwjR"
        >
          <div
            class="fresnel-container fresnel-at-mobile sc-1k44dfv-0 gtzPRh"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 dLObGV"
            >
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 jbBrzI"
              />
            </div>
          </div>
          <div
            class="sc-1k44dfv-0 klHKZs"
            color=""
            style="text-align: center,left;"
          >
            

            <h3
              class="sc-1k44dfv-0 eUGlGC"
            >
              My title
            </h3>
          </div>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 jXxIsO"
          >
            <div
              class="sc-1k44dfv-0 klHKZs"
              color=""
            >
              

              <p>
                My body text
              </p>
            </div>
          </div>
        </div>
        <div
          class="fresnel-container fresnel-greaterThan-mobile sc-1k44dfv-0 gtzPRh"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 dLObGV"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 jbBrzI"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
